### DISCUSSION TITLE
(Displayed fields) Add "Datetime Completed" Column, option to display "Added on"/"Completed on" with hh:mm, maybe also add human readable descriptions

### DISCUSSION THREAD

**Discussion by C84186** [jesec/flood#360] [Ideas]
# More important - Allow display/sorting by "Finished On"

There should be a column to display when a torrent was finished, as well as to sort by that

# Less important - Add more detail/formatting options to datetime columns
 
 ## Options
 
1. It should be possible to see more detail in the datetime column, 
   possibly by:
   * allowing alternate columns that also display date + time (eg ymd-hms, ymd-hm), or perhaps just by time (users can just put them side by side, I assume sorting by either sorts on the underlying time data anyway)
   * allowing users to define format string to control display of datetime:
     * _eg: `%yy-%mm-%dd %HH-%mm`_ 
     * Or whatever is well-defined/conventional- dttm format strings have been done a million time & no doubt there's some convention/standard that can be applied/implemented
1. There should be an option to describe datetime using human-readable conventions, probably as an additional toggle-able column, as described in https://github.com/jesec/flood/discussions/350 

Personally, I'm of the opinion that:
* Allowing manually specified format strings adds extra complexity / a lot of flexibility for an option that will, in practise, only take like, 3 distinct values (eg, date only, date + hour only, date + hour + minute) 
  * Though some USA apologists may want to see `mm-dd-yyyy` or some such nonsense, I'm of the firm opinion that this convention _sucks_ and should be erased from user interfaces. Let them see how we feel when we're subjected to their ridiculous conventions that they impress onto the world at large through sheer force of cultural capital.
* Control over level of detail could be achieved by either:
  * ❓Providing a few more toggle-able columns in UI
    * 📛Trying to provide one for each perturbation is too complex, though
    * ❔ having one for each units *might* be workable (as in, toggle `hours` `minutes` `seconds`.
      * If each unit were broken into its own column, it's probably worth grouping them, similar to how torrent details are locked into diplaying on expanded view
* ❓Format strings - I don't love this idea as it's an entirely new configuration section with its own set of rules for where and how it should be used
* ❔ on the front-end, formatting `dttm` columns depending on how wide the user has chosen to expand the column
   * ✔️ I like this, because it doesnt add complexity to the config, but
   * ❌ Does add a "magic" feature to the frontend that users are expected to discover for themselves. 
   * ❌ Adds an extra level of complexity to client-side rendering logic
   * ❔ Maybe better to:
      *  render the field as its full width, 
      * but default the column width to only fit the date component, &
      * hint that there's more to show with a muted `...` in the field padding or some css pseudo element (so it doesn't widen the field, but indicates that expanding it will increase the level of detail) 
 
 ## Conclusion on my suggested design
 
Between everything that I've put forward, I'd suggest that a combination of:
* adding a few new optional columns:
  * "Time since *event*", &
  * "*event* time-of-day"
  and
* Rendering time information in full detail, but truncating overflow with `...`

**Comment by jesec** [comment]
I think it is a great idea to have those columns. 

I am not sure about the additional customization over the format of dates, though. As I said in #350, the settings panel is already a bit bloated. I think I would have to make some redesigns to accommodate that. 

about the `mm-dd-yyyy`: it depends on your locale. see #123.

**Reply by C84186** [reply]
I personally don't think the settings panel is egregiously bloated, but I agree that just adding them as optional columns is probably the most intuitive & least complex approach.

I'll try to work on this, though I'd appreciate a link in the repo to where to get started
